File: RestReactAspire.Server/Models/Doctor.cs
namespace RestReactAspire.Server.Models;

public class Doctor
{
    public Guid Id { get; set; }
    public required string FirstName { get; set; }
    public required string LastName { get; set; }
    public required string Specialty { get; set; }
    public required string Email { get; set; }
    public required string Phone { get; set; }
}

File: frontend/src/types/doctor.ts
import type { Link, PaginationInfo } from './hateoas';

export interface Doctor {
  id: string;
  firstName: string;
  lastName: string;
  specialty: string;
  email: string;
  phone: string;
  links: Link[];
}

export interface DoctorList {
  items: Doctor[];
  pagination: PaginationInfo;
  links: Link[];
}

export interface CreateDoctorRequest {
  firstName: string;
  lastName: string;
  specialty: string;
  email: string;
  phone: string;
}

export interface UpdateDoctorRequest {
  firstName: string;
  lastName: string;
  specialty: string;
  email: string;
  phone: string;
}
